daemon/Jobs/UpdaterJob.cs

```diff
@@ -67,16 +72,60 @@ public class SpecificVersion
         public int requiredInstallReversion;
     }
 
+    public class DotnetVersion
+    {
+        public Dictionary<string, string> linux;
+        public Dictionary<string, string> windows;
+    }
+
+    public class SourcesDependenciesProperty
+    {
+        public Dictionary<string, DotnetVersion> dotnet;
+        public string nssm { get; set; }
+    }
+
+    public class SourcesInformation
+    {
+        [JsonProperty(PropertyName = "terms-of-service")]
+        public string termsOfService { get; set; }
+
+        public SourcesDependenciesProperty dependencies { get; set; }
+    }
+
+
     public class UpdaterJob : IJob
     {
         // Class Dependencies
         private readonly HttpClient _httpClient;
         private readonly ILogger<UpdaterJob> _logger;
         private readonly AppConfig _config;
-        private readonly Symlink _symlink;
+        private readonly ISymlink _symlink;
         private readonly IApplicationLifetime _applicationLifetime;
+        private readonly IArchitectureUtility _architectureUtility;
+        private readonly IProcessRunner _processRunner;
         private JObject _releaseInformation;
 
+        // Constants
+        private static readonly string[] _validReleaseChannels =
```

Comment:
I'm sure I could work out something today to do this - I just needed something to catch mistakes for then CHANNEL_PLACEHOLDER was defined on accident, and figured this could be a useful safety feature in the future.

Reply:
The problem here is, it stops us from arbitrarily defining/creating channels. And the safety is imo not really needed. Please refactor it to consider `releases.json -> channels` its source of truth for this property.